### PR TITLE
change(v2): swap init parameter order for name and template

### PR DESCRIPTION
**Pull request by wgao19**
## Motivation

Swap init command order for template and project name.

Previously our init command was

```shell
$ init [name] [template]
```

So in the doc we exemplify as 

```shell
$ npx @docusaurus/init@next init my-website classic
```

This is not too bad. But we also accept git url for template, in which case the template argument will be very long and people will nearly certainly copy and paste. 

```shell
$ npx @docusaurus/init@next init [name] https://github.com/wgao19/docusaurus-template-no-style.git
```

Then having template as the last argument is not as friendly as having it before the project name because people can copy and paste the whole thing and input a project name.

```shell
$ npx @docusaurus/init@next init https://github.com/wgao19/docusaurus-template-no-style.git [name]
```

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

NA

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

**Comment by endiliey** [comment]
if we merge this now, need to make sure we release new alpha otherwise the order is wrong for certain version.

id prefer we do a revamp once and for all though. agree on defaulting it to classic template

**Comment by yangshun** [comment]
I agree with Endi's suggestions.

**Comment by wgao19** [comment]
> and also if both options are not passed, we still prompt for name first and then template.

I agree with this, please keep this behavior. But please do not remove the default CLI command argument option. The brevity of just one copy and paste and be done with it is unparalleled. 

Let me rephrase properly what I mean regarding the benefit of swapping the order:

Currently, we instruct people to initialize project like this

```
npx @docusaurus/init@next init [template] [name]
```

This is OK since they're both optional and the template `classic` is quite short and easy to type.

However, if we want to promote initializing projects from git repo, the behavior becomes:

```
npx @docusaurus/init@next init [your project] https://github.com/very-long/git-name-no-one-is-typing
```

People cannot copy and paste this string directly because there is a `[your project]` in the middle.

Then they'd have to copy the first section
```
npx @docusaurus/init@next init 
```

Then type their project name or whatever,

```
[my project]
```

 then copy and paste the git repo

```
https://github.com/very-long/git-name-no-one-is-typing
```

I think the "a lot of benefit" is rather subjective.. it's whether we want to promote the behavior of using repo as template. If we do want that, the difference is very non trivial. Just try it, you copy and paste twice, tab in and out of your terminal, it's ridiculous.

Gatsby's command takes URL in middle doesn't mean it's the best experience.

However, I'm OK with closing this issue because currently both of you are against it. And I'm not very familiar with CLI development so I think I'm not the best person to make the call. I'll thereby close this issue to let you guys plan things properly.
